tracer/src/Datadog.Trace/ITracer.cs

```diff
@@ -36,5 +36,20 @@ public interface ITracer
         /// <param name="settings">Settings for the new <see cref="IScope"/></param>
         /// <returns>A scope wrapping the newly created span</returns>
         IScope StartActive(string operationName, SpanCreationSettings settings);
+
+        /// <summary>
+        /// Sets the details of the user on the local root span
+        /// </summary>
+        /// <param name="email">The user's email</param>
+        /// <param name="name">The user's name</param>
+        /// <param name="id">The user's id</param>
+        /// <param name="sessionId">The user's sessionId</param>
+        /// <param name="role">The user's role</param>
+        public void SetUser(
+            string email = null,
+            string name = null,
+            string id = null,
+            string sessionId = null,
+            string role = null);
```

Comment:
I'm really not a fan of this API. IMO this shouldn't be exposed on the `Tracer`.

Reply:
I wrote similar comments about the API in the RFC: https://docs.google.com/document/d/15dxIrx4D2BM3eUnLNCdMZjN4fQHlzISq31H6RiWRmIM/edit?disco=AAAAVgNvEYU

Reply:
Adding a member to a public interface is a breaking change.

Reply:
Consider using a new type with the 5 fields instead of 5 parameters. This will make things easier later if we need to add more fields.